File: backend/models/interview.py
from typing import List, Optional, Dict, Any
from pydantic import BaseModel, Field
from models.candidate import CandidateProfile

class QuestionDetail(BaseModel):
    question_id: str
    day: int
    module: str
    topic: str
    question: str
    rationale: str
    difficulty: str = "Intermediate"
    is_followup: bool = False
    rag_context: List[str] = Field(default_factory=list)

class TurnEvaluation(BaseModel):
    technical_accuracy: float = 0.0
    communication_score: float = 0.0
    depth_score: float = 0.0
    confidence_score: float = 0.0
    engineering_thinking_score: float = 0.0
    system_design_score: float = 0.0
    composite_score: float = 0.0
    is_followup_needed: bool = False
    feedback_snippet: str = ""
    suggested_probe: Optional[str] = None
    suggested_adaptation: Optional[str] = None

class QuestionTurn(BaseModel):
    turn_index: int
    question: QuestionDetail
    candidate_answer: Optional[str] = None
    evaluation: Optional[TurnEvaluation] = None
    timestamp: str = ""

class InterviewProgress(BaseModel):
    total_questions_asked: int = 0
    days_covered: List[int] = Field(default_factory=list)
    days_count: int = 0
    min_questions_met: bool = False
    min_days_met: bool = False

class StartInterviewRequest(BaseModel):
    candidate_id: str
    target_question_count: int = 8
    min_curriculum_days: int = 4

class SubmitTurnRequest(BaseModel):
    session_id: str
    question_id: str
    candidate_answer: str

class StartInterviewResponse(BaseModel):
    session_id: str
    candidate: CandidateProfile
    status: str = "in_progress"
    current_turn: int = 1
    first_question: QuestionDetail

class SubmitTurnResponse(BaseModel):
    session_id: str
    turn_index: int
    evaluation: TurnEvaluation
    progress: InterviewProgress
    is_interview_complete: bool = False
    next_question: Optional[QuestionDetail] = None

File: frontend/src/types/interview.ts
export interface CandidateProfile {
  id: string;
  name: string;
  target_role: string;
  experience_level: string;
  avatar_color?: string;
  bio: string;
  completed_missions: number[];
  attempts: Record<string, { score: number; pass: boolean }>;
  skipped_topics: number[];
  learning_signals: {
    strongest_domains: string[];
    weakest_domains: string[];
    code_proficiency: string;
    communication_style: string;
    learning_notes: string;
  };
  suggested_focus_days: number[];
}

export interface CurriculumDay {
  day: number;
  module: string;
  title: string;
  summary: string;
  learning_objectives: string[];
  key_concepts: string[];
  tools: string[];
  difficulty: string;
  sample_questions: string[];
}

export interface QuestionDetail {
  question_id: string;
  day: number;
  module: string;
  topic: string;
  question: string;
  rationale: string;
  difficulty: string;
  is_followup: boolean;
  rag_context: string[];
}

export interface TurnEvaluation {
  technical_accuracy: number;
  communication_score: number;
  depth_score: number;
  confidence_score: number;
  engineering_thinking_score: number;
  system_design_score: number;
  composite_score: number;
  is_followup_needed: boolean;
  feedback_snippet: string;
  suggested_probe?: string;
  suggested_adaptation?: string;
}

export interface QuestionTurn {
  turn_index: number;
  question: QuestionDetail;
  candidate_answer: string;
  evaluation?: TurnEvaluation;
  timestamp: string;
}

export interface InterviewProgress {
  total_questions_asked: number;
  days_covered: number[];
  days_count: number;
  min_questions_met: boolean;
  min_days_met: boolean;
}

export interface FeedbackReport {
  session_id: string;
  candidate_id: string;
  candidate_name: string;
  target_role: string;
  total_questions_asked: number;
  curriculum_days_covered: number[];
  overall_score: number;
  technical_knowledge: number;
  communication: number;
  problem_solving: number;
  system_design: number;
  strengths: string[];
  weaknesses: string[];
  missed_topics: string[];
  recommended_revision: string[];
  hiring_recommendation: 'Strong Hire' | 'Hire' | 'Weak Hire' | 'No Hire';
  overall_summary: string;
  charts_data: {
    radar_scores?: Array<{ subject: string; score: number; fullMark?: number }>;
    turn_progression?: Array<{ turn: string; day: string; score: number; accuracy: number }>;
  };
  topic_breakdown: Array<{
    day: number;
    topic: string;
    score: number;
    summary: string;
  }>;
}
